scripts/zones/Wajaom_Woodlands/mobs/Zoraal_Ja_s_Pkuucha.lua

```diff
@@ -11,8 +13,33 @@ function onMobSpawn(mob)
 end;
 
 -----------------------------------
+-- OnMobFight Action
+-----------------------------------
+
+function onMobFight(mob, target)
+    if(mob:getHPP() <= math.random(20,50) and GetMobAction(Percipient_Zoraal_Ja) == ACTION_NONE
```

Comment:
Since onMobFight runs once every second, you should probably set the % to spawn the master at on the spawn, as a local var, then check against that local var instead.